device/core/src/blob_upload/blob_upload_client.ts

```diff
@@ -55,26 +46,26 @@ export interface BlobUpload {
  * @private
  */
 export class BlobUploadClient implements BlobUpload {
+  public clientFileUploadApi: FileUploadInterface;
```

Comment:
>clientFileUploadApi [](start = 9, length = 19)

i don't think there's a need to make this public. you could create the FileUploadApi object in the device client and pass it to the blob client when you instantiate it. then use it directly in the client where it was instantiated.

Reply:
fixed this.